### PR TITLE
Paper UI build process enhanced

### PR DESCRIPTION
**Pull request by hoegertn**
This PR enhances the build process of the PaperUI using injection and concatenation of the JavaScript files. See #800 for further details.

This PR needs #800 to be merged first. I will update this PR and merge the latest changes in, when the other PR is done.

### CODE REVIEW

**Comment by GradyD** [comment]
When building with maven I see this `npm WARN deprecated gulp-minify-html@1.0.5: Consider using gulp-htmlmin instead`. Is there a specific reason you used the deprecated package?

My build failed, not related to your changes, so I will have to finish debugging this over the weekend.

**Comment by hoegertn** [comment]
I am currently using this gulp plugin in my other projects so I know it works. Smarthome is using a rather old version of node, npm, gulp as I see it. I will try to update this and then I will have a look at plugin updates.

Can you send me the output of the gulp maven step? I am wondering why the 'build-main' task is not run

**Comment by sja** [comment]
Hi! As PR #800 is closed, the descreiption here is not up to date. What is the correct order to merge now?

**Comment by hoegertn** [comment]
Hi,
@GradyD is checking this PR.
If this PR works I think it can be merged but it contains the changes of PR #800 
IMHO the reason for closing #800 is superseded by the changes I made in this PR.

tl;dr If this works merge it and leave #800 closed

**Comment by sja** [comment]
So when merging this, we have #800 also inside. Ok, then I'll review this one instead of #800.

**Comment by hoegertn** [comment]
Hi, I just merged master into the PR to resolve the merge conflict

**Comment by GradyD** [comment]
Hey guys,

Just an update, I just opened #829 which updates node and npm.

This should make it easier to move away from bower and also some of the depreciated packages.

**Comment by GradyD** [comment]
Hey guys,

Sorry it took so long. I dug into the error message that I was getting and it was due to my local system having some stuff in NPM cache. Once I cleaned this up I was able to get a successful build. 

I cloned @hoegertn's fork into my eclipse workspace and built openhab2 against it, note I have not updated my IDE to be the new version so I'm still running openhab2 with ESH as my IDE. The UI looks as it should but all calls to /rest/* responded with 404 errors.

When looking at the REST API browser view I get the same error, that /rest/* is not found (404). This could be my environment. The REST plugins I am building with are, 
image and I don't get any errors in the console related to REST.

**Comment by GradyD** [comment]
I did some more testing on this after fixing my /rest and this works without any issues. I do feel that there are some unneeded packages. My biggest concern with this is that the gulp file/packages will start to balloon up instead of being lite and lean. That said I do feel that this has some good improvements added, like combining all of vendor static in vendor.js/css.

**Comment by kaikreuzer** [comment]
Closed as per https://github.com/eclipse/smarthome/pull/873#issuecomment-172865230
